openfisca_france_local/convert_benefit_conditions_to_parameters.py

```diff
@@ -105,10 +105,34 @@ def conditions_to_node_data(conditions: 'list[dict]') -> dict:
 
 def profils_to_node_data(profils: 'list[dict]'):
     def create_profils_field(data: dict, profil: dict):
-        data['profils'].update({profil['type']: {}})
+        if profil['type'] not in data['profils']:
+            data['profils'].update({profil['type']: {}})
 
     def add_profil_with_conditions(data: dict, profil: dict):
-        data['profils'][profil['type']].update(conditions_to_node_data(profil['conditions']))
+        def condition_already_exists_in_node(profil_condition, conditions_in_node_data) -> bool:
+            for condition_type in profil_condition.keys():
```

Comment:
Pas obligé, mais pour facilité la lecture ce que je fais parfois c'est réduire le nombre de branche possibles le plus rapidement possible (1). Par exemple là jusqu'à la toute fin de la méthode on doit gardé en tête qu'est-ce qui se passe si `profil_condition.keys()` est vide. On pourrait mettre un early return pour enlever ce cas des possible. Ça permetrait en plus ligne 118 de return directement `operator in conditions_in_node_data[condition_type]`

(1) oui ça fait deux fois possible :-)

Reply:
Je ne vois pas bien ce que tu veux dire :
- si `profil_condition.keys()` est vide, on ne rentre pas dans le for
- Je peux déjà return comme tu le dis en ligne 118 (et merci, je l'ai fait)

Reply:
Justement, mais là on doit garder en tête que si c'est vide faudrait regarder **après** le `for`, alors que si on fait avant le for 
```
if len(profil_condition) == 0:
    return False
```

On est sûr à 100% qu'on rentre dans le `for`

Reply:
Encore une fois pas obligé, voici un blog post qui parle entre autre de cet argument.

https://freedium.cfd/https://betterprogramming.pub/are-early-returns-any-good-eed4b4d03866

> The last argument that I'm going to make for using early returns is peace of mind for the programmer. By using early returns, you get invalid cases out of the way first (bouncer pattern), put a blank line in there next, and then you can focus on the "real" body of the function.

Reply:
D'accord, je comprends!
C'est un article intéressant et je vois maintenant l'intérêt pour le cas ici, je trouve effectivement que c'est pas plus mal de rendre ce code un peu plus facile à lire et je pense que c'est le cas avec ce "early return"
Merci!